sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/SpannerIO.java

```diff
@@ -1496,8 +1505,10 @@ public ReadChangeStream withTraceSampleProbability(Double probability) {
               getMetadataInstance(), changeStreamDatabaseId.getInstanceId().getInstance());
       final String partitionMetadataDatabaseId =
           MoreObjects.firstNonNull(getMetadataDatabase(), changeStreamDatabaseId.getDatabase());
-      final String partitionMetadataTableName =
-          generatePartitionMetadataTableName(partitionMetadataDatabaseId);
+      String partitionMetadataTableName = getMetadataTableName();
```

Comment:
Maybe a bit simpler like the lines above?
`MoreObjects.firstNonNull(getMetadataDatabase(), generatePartitionMetadataTableName(partitionMetadataDatabaseId));`

Reply:
Good catch.